packages/kit/src/exports/vite/index.js

```diff
@@ -306,7 +306,13 @@ async function kit({ svelte_config }) {
 						//    See https://github.com/sveltejs/kit/pull/9172
 						//    and https://vitest.dev/config/#deps-registernodeloader
 						'@sveltejs/kit'
-					]
+					],
+					resolve: {
+						// This is default value in Vite 6 but not Vite 5. We need it so that
+						// importing $app/state correctly resolves to the default conditional export
+						// instead of the browser condition.
+						conditions: ['module', 'node', 'development|production']
```

Comment:
```suggestion
						// We need this so that importing $app/state correctly
						// resolves to the default conditional export instead of
						// the browser condition on the server.
						conditions: ['module', 'node']
```